Experimental prototype for converting torch.jit.trace modules to export

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #124449

Differential Revision: [D56440613](https://our.internmc.facebook.com/intern/diff/D56440613)

We want to do this for following reasons:
1. There is current limitation in export tracing for torch.jit.trace d modules that cannot be easily upstreamed
2. We need to run internal CI regularly to understand feature gaps and continuously track them
3. Multiple people will be working on this prototype so it is better to have a checked in version so we don't always run into merge conflicts. 

cc @ezyang @msaroufim @bdhirsh @anijain2305 @chauhang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng

tugsbayasgalan committed:
Hacks to work around the fact that ScriptMethod does not have code/signature

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56440613).

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56440613).

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56440613).

tugsbayasgalan committed:
Update on "Hacks to work around the fact that ScriptMethod does not have code/signature"


Differential Revision: [D56440613](https://our.internmc.facebook.com/intern/diff/D56440613)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Hacks to work around the fact that ScriptMethod does not have code/signature"


Differential Revision: [D56440613](https://our.internmc.facebook.com/intern/diff/D56440613)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Hacks to work around the fact that ScriptMethod does not have code/signature"


Differential Revision: [D56440613](https://our.internmc.facebook.com/intern/diff/D56440613)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Hacks to work around the fact that ScriptMethod does not have code/signature"


Differential Revision: [D56440613](https://our.internmc.facebook.com/intern/diff/D56440613)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Hacks to work around the fact that ScriptMethod does not have code/signature"


Differential Revision: [D56440613](https://our.internmc.facebook.com/intern/diff/D56440613)

cc ezyang msaroufim bdhirsh anijain2305 chauhang voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56440613).

tugsbayasgalan committed:
Update on "Hacks to work around the fact that ScriptMethod does not have code/signature"


Differential Revision: [D56440613](https://our.internmc.facebook.com/intern/diff/D56440613)

cc ezyang msaroufim bdhirsh anijain2305 chauhang voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56440613).

tugsbayasgalan committed:
Update on "Hacks to work around the fact that ScriptMethod does not have code/signature"


Differential Revision: [D56440613](https://our.internmc.facebook.com/intern/diff/D56440613)

cc ezyang msaroufim bdhirsh anijain2305 chauhang voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Experimental prototype for converting torch.jit.trace modules to export"


Differential Revision: [D56440613](https://our.internmc.facebook.com/intern/diff/D56440613)

cc ezyang msaroufim bdhirsh anijain2305 chauhang voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Experimental prototype for converting torch.jit.trace modules to export"


Differential Revision: [D56440613](https://our.internmc.facebook.com/intern/diff/D56440613)

cc ezyang msaroufim bdhirsh anijain2305 chauhang voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Experimental prototype for converting torch.jit.trace modules to export"


Differential Revision: [D56440613](https://our.internmc.facebook.com/intern/diff/D56440613)

We want to do this for following reasons:
1. There is current limitation in export tracing for torch.jit.trace d modules that cannot be easily upstreamed
2. We need to run internal CI regularly to understand feature gaps and continuously track them
3. Multiple people will be working on this prototype so it is better to have a checked in version so we don't always run into merge conflicts. 

cc ezyang msaroufim bdhirsh anijain2305 chauhang voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56440613).

tugsbayasgalan committed:
Update on "Experimental prototype for converting torch.jit.trace modules to export"


Differential Revision: [D56440613](https://our.internmc.facebook.com/intern/diff/D56440613)

We want to do this for following reasons:
1. There is current limitation in export tracing for torch.jit.trace d modules that cannot be easily upstreamed
2. We need to run internal CI regularly to understand feature gaps and continuously track them
3. Multiple people will be working on this prototype so it is better to have a checked in version so we don't always run into merge conflicts. 

cc ezyang msaroufim bdhirsh anijain2305 chauhang voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56440613).

angelayi commented on torch/_export/non_strict_utils.py:
```diff
@@ -153,11 +148,22 @@ def make_fake_inputs(nn_module, args, kwargs, dynamic_shapes):
         ), "Found constraints when tracing with a toplevel tracing context."
         fake_mode = context.fake_mode
-    else:
+    elif not _is_torch_jit_trace:
```
can we leave a comment like "HACK"

tugsbayasgalan committed:
Update on "Experimental prototype for converting torch.jit.trace modules to export"


Differential Revision: [D56440613](https://our.internmc.facebook.com/intern/diff/D56440613)

We want to do this for following reasons:
1. There is current limitation in export tracing for torch.jit.trace d modules that cannot be easily upstreamed
2. We need to run internal CI regularly to understand feature gaps and continuously track them
3. Multiple people will be working on this prototype so it is better to have a checked in version so we don't always run into merge conflicts. 

cc ezyang msaroufim bdhirsh anijain2305 chauhang voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56440613).

avikchaudhuri commented on torch/_export/non_strict_utils.py:
```diff
@@ -259,9 +270,13 @@ def produce_guards_and_solve_constraints(
+        )
+    else:
+        # FIXME(ycao): This is a hack to get around missing signature from ScriptMethod
+        msg = "dummy constraint violation message"
```
Wonder if it's better to just have `msg = ""` here, analogous to other zero-d out things in this mode. It might even be better to have a top-level dict of all the things we're zero-ing out, something like `torch_jit_trace_config = {"msg": "", "original_signature": None, ...}` etc. so that we have a good account of all the FIXMEs.
